packages/fuels-types/src/parse_param.rs

```diff
@@ -29,12 +29,20 @@ impl ParamType {
                     // Try to parse tuple (T, T, ..., T)
                     return ParamType::parse_tuple_param(prop, types);
                 }
+                if prop.type_field.contains("generic") {
+                    return Ok(Self::parse_generic_param(prop));
+                }
                 // Try to parse a free form enum or struct (e.g. `struct MySTruct`, `enum MyEnum`).
                 ParamType::parse_custom_type_param(prop, types)
             }
         }
     }
 
+    fn parse_generic_param(prop: &TypeDeclaration) -> ParamType {
+        let name = prop.type_field.split("generic").collect::<Vec<&str>>()[1].trim();
```

Comment:
Friendly boundary check / error handling:

```suggestion
        let name = prop
            .type_field
            .split("generic")
            .collect::<Vec<&str>>()
            .get(1)
            .expect("generic param string not properly formatted. This should never happen.")
            .trim();
```

Reply:
Good point. I've taken this into consideration in the rewrite of this part, take a look when you can.